dcpy/__init__.py

```diff
@@ -0,0 +1,3 @@
+from pydantic import BaseModel
+
+BaseModel.model_config["coerce_numbers_to_str"] = True
```

Comment:
So this sets this globally. Open to thoughts on that, another solution would be to have `version: Annotated[...]` rather than `version: str` in the recipe model. But I really can't think of a case where we have a `str` field and wouldn't want to allow coercion from an int/float. We're pretty exclusively reading in models from yml.

@alexrichey @damonmcc @sf-dcp would be interested in thoughts

Reply:
For context - in pydantic 1, `version: 20240101` was fine in a recipe.yml, with version in our python class being a string. In v2, this throws an error.